pkg/runner/step_test.go

```diff
@@ -0,0 +1,276 @@
+			name:   "testMergeIntoEmptyMap",
+			target: map[string]string{},
+			maps: []map[string]string{
+				{
+					"key1": "value1",
+					"key2": "value2",
+				}, {
+					"key2": "overridden",
+					"key3": "value3",
+				},
+			},
+			expected: map[string]string{
+				"key1": "value1",
+				"key2": "overridden",
+				"key3": "value3",
+			},
+		},
+		{
+			name: "testMergeIntoExistingMap",
+			target: map[string]string{
+				"key1": "value1",
+				"key2": "value2",
+			},
+			maps: []map[string]string{
+				{
+					"key1": "overridden",
+				},
+			},
+			expected: map[string]string{
+				"key1": "overridden",
+				"key2": "value2",
+			},
+		},
+	}
+
+	for _, tt := range table {
+		t.Run(tt.name, func(t *testing.T) {
+			mergeIntoMap(&tt.target, tt.maps...)
+			assert.Equal(t, tt.expected, tt.target)
+		})
+	}
+}
+
+type stepMock struct {
+	mock.Mock
+	step
+}
+
+func (sm *stepMock) pre() common.Executor {
+	args := sm.Called()
+	return args.Get(0).(func(context.Context) error)
+}
+
+func (sm *stepMock) main() common.Executor {
+	args := sm.Called()
+	return args.Get(0).(func(context.Context) error)
+}
+
+func (sm *stepMock) post() common.Executor {
+	args := sm.Called()
+	return args.Get(0).(func(context.Context) error)
+}
+
+func (sm *stepMock) getRunContext() *RunContext {
+	args := sm.Called()
+	return args.Get(0).(*RunContext)
+}
+
+func (sm *stepMock) getStepModel() *model.Step {
+	args := sm.Called()
+	return args.Get(0).(*model.Step)
+}
+
+func (sm *stepMock) getEnv() *map[string]string {
+	args := sm.Called()
+	return args.Get(0).(*map[string]string)
+}
+
+func TestSetupEnv(t *testing.T) {
+	cm := &containerMock{}
+	sm := &stepMock{}
+
+	rc := &RunContext{
+		Config: &Config{
+			Env: map[string]string{
+				"GITHUB_RUN_ID": "runId",
+			},
+		},
+		Run: &model.Run{
+			JobID: "1",
+			Workflow: &model.Workflow{
+				Jobs: map[string]*model.Job{
+					"1": {
+						Env: yaml.Node{
+							Value: "JOB_KEY: jobvalue",
+						},
+					},
+				},
+			},
+		},
+		Env: map[string]string{
+			"RC_KEY": "rcvalue",
+		},
+		ExtraPath:    []string{"/path/to/extra/file"},
+		JobContainer: cm,
+	}
+	step := &model.Step{
+		With: map[string]string{
+			"STEP_WITH": "with-value",
+		},
+	}
+	env := map[string]string{
+		"PATH": "",
+	}
+
+	sm.On("getRunContext").Return(rc)
+	sm.On("getStepModel").Return(step)
+	sm.On("getEnv").Return(&env)
+
+	cm.On("UpdateFromImageEnv", &env).Return(func(ctx context.Context) error { return nil })
+	cm.On("UpdateFromEnv", "/var/run/act/workflow/envs.txt", &env).Return(func(ctx context.Context) error { return nil })
+	cm.On("UpdateFromPath", &env).Return(func(ctx context.Context) error { return nil })
+
+	err := setupEnv(context.Background(), sm)
+	assert.Nil(t, err)
+
+	// These are commit or system specific
+	delete((env), "GITHUB_REF")
+	delete((env), "GITHUB_SHA")
+	delete((env), "GITHUB_WORKSPACE")
+	delete((env), "GITHUB_REPOSITORY")
+	delete((env), "GITHUB_REPOSITORY_OWNER")
+	delete((env), "GITHUB_ACTOR")
```

Comment:
`GITHUB_ACTOR` is hardcoded to `nektos/act`, so doesn't need to be excluded here.

Reply:
I would still like to exclude it, since it could be overridden by env-vars.
So it is out of control of this test (kind of).